File: Model/src/main/java/com/promotick/lafabril/model/reporte/ReporteMundialPolla.java
package com.promotick.lafabril.model.reporte;

import com.promotick.apiclient.utils.file.excel.annotations.ExcelClass;
import com.promotick.apiclient.utils.file.excel.annotations.ExcelColumn;
import lombok.Data;

import java.io.Serializable;

@Data
@ExcelClass(filename = "Reporte-mundial-polla", sheetName = "Reporte Mundial Polla")
public class ReporteMundialPolla implements Serializable{

	private static final long serialVersionUID = 2068885050024304718L;

	private Integer idParticipante;

	@ExcelColumn("Catalogo")
	private String nombreCatalogo;

	@ExcelColumn("Usuario")
	private String usuarioParticipante;

	@ExcelColumn("Nombre")
	private String nombreParticipante;

	@ExcelColumn("Apellido Paterno")
	private String appaterno;

	@ExcelColumn("Apellido Materno")
	private String apmaterno;

	@ExcelColumn("Broker")
	private String broker;

	@ExcelColumn("Ciudad")
	private String ciudad;

	@ExcelColumn("Regional")
	private String regional;

	@ExcelColumn("Estado")
	private String statusParticipante;

	@ExcelColumn("Fecha Respuesta")
	private String fechaRespuesta;

	@ExcelColumn("Hora Respuesta")
	private String horaRespuesta;

	@ExcelColumn("Completado")
	private String completado;

	@ExcelColumn("Grupo A1")
	private String a1;

	@ExcelColumn("Grupo A2")
	private String a2;

	@ExcelColumn("Grupo B1")
	private String b1;

	@ExcelColumn("Grupo B2")
	private String b2;

	@ExcelColumn("Grupo C1")
	private String c1;

	@ExcelColumn("Grupo C2")
	private String c2;

	@ExcelColumn("Grupo D1")
	private String d1;

	@ExcelColumn("Grupo D2")
	private String d2;

	@ExcelColumn("Grupo E1")
	private String e1;

	@ExcelColumn("Grupo E2")
	private String e2;
	@ExcelColumn("Grupo F1")
	private String f1;
	@ExcelColumn("Grupo F2")
	private String f2;
	@ExcelColumn("Grupo G1")
	private String g1;
	@ExcelColumn("Grupo G2")
	private String g2;
	@ExcelColumn("Grupo H1")
	private String h1;
	@ExcelColumn("Grupo H2")
	private String h2;
	@ExcelColumn("1A2B")
	private String a1b2;
	@ExcelColumn("1B2A")
	private String b1a2;
	@ExcelColumn("1C2D")
	private String c1d2;
	@ExcelColumn("1D2C")
	private String d1c2;
	@ExcelColumn("1E2F")
	private String e1f2;
	@ExcelColumn("1F2E")
	private String f1e2;
	@ExcelColumn("1G2H")
	private String g1h2;
	@ExcelColumn("1H2G")
	private String h1g2;
	@ExcelColumn("1A2B / 1C2D")
	private String a1b2c1d2;
	@ExcelColumn("1B2A / 1D2C")
	private String b1a2d1c2;
	@ExcelColumn("1E2F / 1G2H")
	private String e1f2g1h2;
	@ExcelColumn("1F2E / 1H2G")
	private String f1e2h1g2;
	@ExcelColumn("1A2B / 1C2D - 1E2F / 1G2H")
	private String a1b2c1d2e1f2g1h2;
	@ExcelColumn("1B2A / 1D2C - 1F2E / 1H2G")
	private String b1a2d1c2f1e2h1g2;
	@ExcelColumn("FINAL")
	private String finalPartido;
}

File: Resources/WebContent/public/admin/js/views/reporte-mundial-polla.js
function ReporteMundialPolla(uri) {
    this.uri = uri;
    this.rangoDatePicker = $('#rangoDatePicker');
    this.grid = $('#grid');
    this.btnFiltrar = $('#btnFiltrar');
    this.fechaInicio = $('#fechaInicio');
    this.fechaFin = $('#fechaFin');
    this.btnDescargar = $('#btnDescargar');
    this.inptBuscar = $('#buscarParticipante');
    this.btnBuscar = $('#btnBuscar');
}

ReporteMundialPolla.prototype.init = function() {
    this.handle();
    this.listar();
};

ReporteMundialPolla.prototype.handle = function() {
    var obj = this;

    obj.rangoDatePicker.datepicker({
        endDate: "0m"
    });

    obj.btnFiltrar.on('click', function(e){
        e.preventDefault();
        obj.gridReload();
    });

    obj.btnBuscar.on('click', function(e){
        e.preventDefault();
        obj.gridReload();
    });

    obj.btnDescargar.on('click', function(e) {
        e.preventDefault();
        window.location.href = $(this).attr('data-descarga');
    })

};

ReporteMundialPolla.prototype.listar = function(){
    var obj = this;
    obj.grid.DataTable({
        "bProcessing" : true,
        "bFilter" : false,
        "bLengthChange": false,
        "bPaginate": true,
        "bInfo" : true,
        "serverSide" : true,
        "bSort": false,
        "ajax": {
            "url" : obj.uri + 'reportes/mundial-polla/listar',
            "type": "POST",
            "data" : function(d) {
                d.nroDocumento = obj.inptBuscar.val();
                // d.finicio = obj.fechaInicio.data('datepicker').getFormattedDate('yyyy-mm-dd');
                // d.ffin = obj.fechaFin.data('datepicker').getFormattedDate('yyyy-mm-dd');
            }
        },
        "columns" : [
            {"data" : "nombreCatalogo", "orderable": false},
            {"data" : "usuarioParticipante", "orderable": false},
            {"data" : "nombreParticipante", "orderable": false},
            {"data" : "appaterno", "orderable": false},
            {"data" : "apmaterno", "orderable": false},
            {"data" : "broker", "orderable": false},
            {"data" : "ciudad", "orderable": false},
            {"data" : "regional", "orderable": false},
            {"data" : "statusParticipante", "orderable": false},
            {"data" : "fechaRespuesta", "orderable": false, "render": this._formatNull},
            {"data" : "horaRespuesta", "orderable": false, "render": this._formatNull},
            {"data" : "completado", "orderable": false, "render": this._formatNull},
            {"data" : "a1", "orderable": false, "render": this._formatNR},
            {"data" : "a2", "orderable": false, "render": this._formatNR},
            {"data" : "b1", "orderable": false, "render": this._formatNR},
            {"data" : "b2", "orderable": false, "render": this._formatNR},
            {"data" : "c1", "orderable": false, "render": this._formatNR},
            {"data" : "c2", "orderable": false, "render": this._formatNR},
            {"data" : "d1", "orderable": false, "render": this._formatNR},
            {"data" : "d2", "orderable": false, "render": this._formatNR},
            {"data" : "e1", "orderable": false, "render": this._formatNR},
            {"data" : "e2", "orderable": false, "render": this._formatNR},
            {"data" : "f1", "orderable": false, "render": this._formatNR},
            {"data" : "f2", "orderable": false, "render": this._formatNR},
            {"data" : "g1", "orderable": false, "render": this._formatNR},
            {"data" : "g2", "orderable": false, "render": this._formatNR},
            {"data" : "h1", "orderable": false, "render": this._formatNR},
            {"data" : "h2", "orderable": false, "render": this._formatNR},
            {"data" : "a1b2", "orderable": false, "render": this._formatNR},
            {"data" : "b1a2", "orderable": false, "render": this._formatNR},
            {"data" : "c1d2", "orderable": false, "render": this._formatNR},
            {"data" : "d1c2", "orderable": false, "render": this._formatNR},
            {"data" : "e1f2", "orderable": false, "render": this._formatNR},
            {"data" : "f1e2", "orderable": false, "render": this._formatNR},
            {"data" : "g1h2", "orderable": false, "render": this._formatNR},
            {"data" : "h1g2", "orderable": false, "render": this._formatNR},
            {"data" : "a1b2c1d2", "orderable": false, "render": this._formatNR},
            {"data" : "b1a2d1c2", "orderable": false, "render": this._formatNR},
            {"data" : "e1f2g1h2", "orderable": false, "render": this._formatNR},
            {"data" : "f1e2h1g2", "orderable": false, "render": this._formatNR},
            {"data" : "a1b2c1d2e1f2g1h2", "orderable": false, "render": this._formatNR},
            {"data" : "b1a2d1c2f1e2h1g2", "orderable": false, "render": this._formatNR},
            {"data" : "finalPartido", "orderable": false, "render": this._formatNR},


        ],
        "lengthMenu": [[10, 15, 20], [10, 15, 20]],
        "initComplete":function(settings,json){

        },
        "fnDrawCallback": function(){

        }
    });
};

ReporteMundialPolla.prototype.gridReload = function(){
    this.grid.DataTable().ajax.reload();
};

ReporteMundialPolla.prototype._formatArchivo = function (data, type, full) {
    return '<a href="' + data + '" target="_blank" class="btn btn-sm btn-success"><i class="fa fa-download" aria-hidden="true"></i> Descargar</a>';
};

ReporteMundialPolla.prototype._formatNull = function(data, type, full) {
    return data != null ? data : '-';
};

ReporteMundialPolla.prototype._formatNR = function(data, type, full) {
    return data != null ? data : '-';
};